src/fhirtypes/ElementDefinition.ts

```diff
@@ -1699,27 +1699,59 @@ export class ElementDefinition {
     ) {
       let newElements: ElementDefinition[] = [];
       if (this.contentReference) {
-        // If the element is a content reference, we need to unfold from the referenced element
+        // If the element is a content reference, we need to unfold from the unconstrained referenced element and then apply any
+        // extensions from the profile.  So far, extensions are the only thing we've been able to confirm should be carried over.
+        // See: https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Clarification.20on.20contentReference
+
         // Get the original resource JSON so we unfold unconstrained reference
         const type = this.structDef.type;
         const json = fisher.fishForFHIR(type, Type.Resource);
         if (json) {
           const def = StructureDefinition.fromJSON(json);
           // Content references start with #, slice that off to id of referenced element
-          const referencedElement = def.findElement(
-            this.contentReference.slice(this.contentReference.indexOf('#') + 1)
-          );
+          const contentRefId = this.getContentReferenceId();
+          const referencedElement = def.findElement(contentRefId);
+          const recursiveContentReferencePaths: string[] = [];
           newElements = referencedElement?.children().map(e => {
             const eClone = e.clone();
             eClone.id = eClone.id.replace(referencedElement.id, this.id);
             eClone.structDef = this.structDef;
             eClone.captureOriginal();
+            if (e.getContentReferenceId() === contentRefId) {
+              // This is a nested recursive content reference.  We'll want to avoid this when carrying over elements from the profile.
+              recursiveContentReferencePaths.push(e.path);
+            }
             return eClone;
           });
           if (newElements.length > 0) {
             // If we successfully unfolded, this element is no longer a content reference
             this.type = referencedElement.type;
             delete this.contentReference;
+
+            // Now carry over the extension elements (and their children) from the profile
+            const elementsToCarryOver = this.structDef
+              ?.findElement(contentRefId)
+              ?.children()
+              ?.filter(e => {
+                return (
+                  // an extension of child of extension under the unrolled path
+                  /(^|\.)(modifierE|e)xtension([.:]|$)/.test(e.id.slice(contentRefId.length + 1)) &&
+                  // but not under a recursive content reference because that would be a real mess!
+                  !recursiveContentReferencePaths.some(p => e.path.startsWith(`${p}.`))
```

Comment:
I was trying to understand the case where we would filter out these recursive content reference paths, which led me to try taking out this line, which made me notice that it doesn't look like either test includes a check to ensure these are filtered out properly. Does it make sense to add that to one of the tests or add a new test? Or am is this too much of an edge case?

Reply:
I'll try to cover it in a test.